src/main/scala/beam/sim/vehiclesharing/AvailabilityBehaviorBasedRepositioning.scala

```diff
@@ -0,0 +1,157 @@
+      val availValMin = availVal.drop(1).foldLeft(availVal.headOption.map(_.observations).getOrElse(0)) { (minV, cur) =>
+        Math.min(minV, cur.observations)
+      }
+      minAvailabilityMap.put((i, taz.tazId), availValMin)
+      val inquiryVal =
+        getCollectedDataFromPreviousSimulation(time, taz.tazId, RepositionManager.inquiry).map(_.observations).sum
+      val boardingVal =
+        getCollectedDataFromPreviousSimulation(time, taz.tazId, RepositionManager.boarded).map(_.observations).sum
+      unboardedVehicleInquiry.put((i, taz.tazId), inquiryVal - boardingVal)
+    }
+  }
+
+  def getCollectedDataFromPreviousSimulation(
+                                              time: Int,
+                                              idTAZ: Id[TAZ],
+                                              label: String
+                                            ): Vector[TAZSkimmerInternal] = {
+    val fromBin = time / statTimeBin
+    val untilBin = (time + repositionTimeBin) / statTimeBin
+    (fromBin until untilBin)
+      .map(i => beamServices.skims.taz_skimmer.getLatestSkim(i, idTAZ, vehicleManager.toString, label))
+      .toVector
+      .flatten
+  }
+
+  override def getVehiclesForReposition(
+                                         now: Int,
+                                         timeBin: Int,
+                                         availableFleet: List[BeamVehicle]
+                                       ): List[(BeamVehicle, SpaceTime, Id[TAZ], SpaceTime, Id[TAZ])] = {
+
+    val oversuppliedTAZ = mutable.TreeSet.empty[RepositioningRequest](orderingAvailVeh)
+    val undersuppliedTAZ = mutable.TreeSet.empty[RepositioningRequest](orderingShortage)
+
+    val nowRepBin = now / timeBin
+    val futureRepBin = nowRepBin + 1
+    beamServices.beamScenario.tazTreeMap.getTAZs.foreach { taz =>
+      val availValMin = minAvailabilityMap((nowRepBin, taz.tazId))
+      val InquiryUnboarded = unboardedVehicleInquiry((futureRepBin, taz.tazId))
+      if (availValMin > 0) {
+        oversuppliedTAZ.add(RepositioningRequest(taz, availValMin, 0))
+      } else if (InquiryUnboarded > 0) {
+        undersuppliedTAZ.add(RepositioningRequest(taz, 0, InquiryUnboarded))
+      }
+    }
+
+    val topOversuppliedTAZ = oversuppliedTAZ.take(matchLimit)
+    val topUndersuppliedTAZ = undersuppliedTAZ.take(matchLimit)
+    val ODs = new mutable.ListBuffer[(RepositioningRequest, RepositioningRequest, Int, Int)]
+    while (topOversuppliedTAZ.nonEmpty && topUndersuppliedTAZ.nonEmpty) {
+      @SuppressWarnings(Array("UnsafeTraversableMethods"))
+      val org = topOversuppliedTAZ.head
+      var destTimeOpt: Option[(RepositioningRequest, Int)] = None
+      topUndersuppliedTAZ.foreach { dst =>
+        val vehicleTypeId =
+          Id.create( // FIXME Vehicle type borrowed from ridehail -- pass the vehicle type of the car sharing fleet instead
```

Comment:
Can this be fixed before the PR is done? Otherwise it should be an issue, as FIXMEs are often left to rot.

Reply:
yup